tests/frontier/opcodes/test_all_opcodes.py

```diff
@@ -0,0 +1,106 @@
+"""
+Call every possible opcode to see if it is recognized by the evm
+If opcode is undefined the subcall will fail
+"""
+
+from typing import Dict
+
+import pytest
+
+from ethereum_test_forks import Fork
+from ethereum_test_tools import (
+    Account,
+    Address,
+    Alloc,
+    Bytecode,
+    Environment,
+    StateTestFiller,
+    Transaction,
+)
+from ethereum_test_tools.vm.opcode import Opcode
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+from ethereum_test_tools.vm.opcode import UndefinedOpcodes
+
+REFERENCE_SPEC_GIT_PATH = "N/A"
+REFERENCE_SPEC_VERSION = "N/A"
+all_opcodes = set(Op)
+undefined_opcodes = set(UndefinedOpcodes)
+
+
+def prepare_stack(opcode: Opcode) -> Bytecode:
+    """Prepare valid stack for opcode"""
+    if opcode == Op.CREATE:
+        return Op.MSTORE(0, 0x6001600155) + Op.PUSH1(5) + Op.PUSH1(27) + Op.PUSH1(5)
+    if opcode == Op.CREATE2:
+        return Op.MSTORE(0, 0x6001600155) + Op.PUSH1(1) + Op.PUSH1(5) + Op.PUSH1(27) + Op.PUSH1(5)
+    if opcode == Op.JUMPI:
+        return Op.PUSH1(1) + Op.PUSH1(5)
+    if opcode == Op.JUMP:
+        return Op.PUSH1(3)
+    return Op.PUSH1(0x01) * 32
+
+
+def prepare_suffix(opcode: Opcode) -> Bytecode:
+    """Prepare after opcode instructions"""
+    if opcode == Op.JUMPI or opcode == Op.JUMP:
+        return Op.JUMPDEST
+    return Op.STOP
+
+
+@pytest.mark.valid_from("Frontier")
+def test_all_opcodes(state_test: StateTestFiller, pre: Alloc, fork: Fork):
+    """Check if the opcode supported on given fork"""
+    code_worked = 1000
+
+    code_contract: Dict[Opcode, Address] = {}
+    for opcode in sorted(all_opcodes | undefined_opcodes):
+        code_contract[opcode] = pre.deploy_contract(
```

Comment:
Might be getting ahead of myself, but this code does not handle EOF opcodes, and it's possible that we need more information inside of each opcode for this to be seamless.

We could add a new property to [`Bytecode` class](https://github.com/ethereum/execution-spec-tests/blob/14f9cbf32d3e69e9ca18d9f8c0ef80d526277f7b/src/ethereum_test_vm/bytecode.py#L9) like `supported_evm_code_types` which could contain a set of [`EVMCodeType`](https://github.com/ethereum/execution-spec-tests/blob/14f9cbf32d3e69e9ca18d9f8c0ef80d526277f7b/src/ethereum_test_vm/evm_types.py#L9C7-L9C18) and e.g. `Op.PUSH1.supported_evm_code_types == {EVMCodeType.LEGACY, EVMCodeType.EOF_V1}`, `Op.EXTCALL.supported_evm_code_types == {EVMCodeType.EOF_V1}` and `Op.CALL.supported_evm_code_types == {EVMCodeType.LEGACY}`.

Then in this case we read that property and deploy a different contract per each evm code type.

We could leave this for a follow-up PR because it seems out-of-scope in my opinion.

Reply:
All opcodes in eof already covered in eof tests.

This is frontier model test. And it also need to ensure that eof opcodes rejected in legacy code

Reply:
I think this is would be a useful addition to `Opcodes`, regardless of whether we apply it in this test or not. Created a follow-up issue #770.